core/src/main/java/hudson/cli/ReloadConfigurationCommand.java

```diff
@@ -43,8 +46,26 @@ public String getShortDescription() {
 
     @Override
     protected int run() throws Exception {
-        Jenkins.getActiveInstance().doReload();
-        return 0;
+        Jenkins j = Jenkins.getInstance();
+        // Or perhaps simpler to inline the thread body of doReload?
+        j.doReload();
+        Object app;
+        while ((app = WebApp.get(j.servletContext).getApp()) instanceof HudsonIsLoading) {
+            Thread.sleep(100);
```

Comment:
Well, if the command hangs then someone will kill it. I see no need for an internal timeout.